lib/ecto/adapters/myxql.ex

```diff
@@ -238,7 +238,7 @@ defmodule Ecto.Adapters.MyXQL do
         {:ok, last_insert_id(key, last_insert_id)}
 
       {:error, err} ->
-        case @conn.to_constraints(err) do
+        case @conn.to_constraints(err, [source: sql]) do
```

Comment:
this is interesting, the default implementation provided by Ecto.Adapters.SQL calls to_constraints/1: https://github.com/elixir-ecto/ecto_sql/blob/v3.3.2/lib/ecto/adapters/sql.ex#L681

but here we're using a custom implementation specific to MyXQL. So the question is should we add `@callback to_constraints(error, opts)` after all? I'm not sure, maybe we should wait for more use cases and meanwhile have:

```elixir
defmodule Ecto.Adapters.MyXQL.Connection do
  @impl true
  def to_constraints(error) do
    to_constraints(error, [])
  end

  @doc false
  def to_constraints(error, opts) do
    ...
  end
end
```

wdyt @josevalim?

Reply:
spoken with José, we are going to go with code in my comment.

Reply:
Nice. I have made the change and pushed a new version.